### ISSUE TITLE
id_ref Serialization - invalid type

### ISSUE THREAD

**Issue by cryptoquick** [kardeiz/jsonrpc-v2#16]
When I go to issue a request:

```json
{"jsonrpc": "2.0", "method":"My.Rpc.Method", "params": [], "id": 0}
```

When parsing that message string into a RequestObject using serde_json:

```rust
match serde_json::from_str(&request_text)
    as Result<jsonrpc_v2::RequestObject, serde_json::Error>
{
    Ok(call) => match &*call.method_ref() {
        MY_RPC_METHOD_CONST => {
```

Which is parsed into:

```rust
RequestObject { jsonrpc: V2, method: "My.Rpc.Method", params: Some(Value(Array([]))), id: Some(Some(Num(0))) }
```

I then want to get the id, otherwise, it's Null:

```rust
let request_id = match call.id_ref() {
  Some(id) => id.to_owned(),
  None => jsonrpc_v2::Id::Null,
};
```

Then, when building the RPC request:

```rust
jsonrpc_v2::RequestObject::request()
  .with_method(MY_RPC_METHOD_CONST)
  .with_id(request_id.clone())
  .finish()
```

This returns the following error:

`thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: invalid type: integer `0`, expected variant identifier'`

Not sure how to get around this?

**Comment by kardeiz** [comment]
@cryptoquick I am not able to replicate this issue:

```rust
let req: RequestObject = serde_json::from_str(r#"{"jsonrpc": "2.0", "method":"My.Rpc.Method", "params": [], "id": 0}"#).unwrap();

let other_id = match req.id_ref() {
    Some(id) => id.to_owned(),
    None => Id::Null,
};

let req2 = RequestObject::request()
    .with_method("sub")
    .with_id(other_id.clone())
    .with_params(serde_json::json!([12, 2]))
    .finish();

let res = rpc.handle(req2).await;

println!("{}", serde_json::to_string_pretty(&res)?);
```

Behaves appropriately, using the same ID as the first request.

If you can set up a minimal reproducible example, I would be happy to investigate further.

Also, there are no `unwrap`s in the `jsonrpc_v2` library. Does your panic contain a line number and file?

**Comment by cryptoquick** [comment]
No kidding! That's a good point, I'll check to see if this is more in my code than yours. I created a bit of a helper utility function to make RPC calls after building them, and that piece I omitted because I thought it wasn't related. And now that I think about it, the unwrap was on the call to that function...

If I keep bashing my head against this, I'll definitely set up an example to repro... Otherwise, I think your intuition that there's something else at play might be correct...
